crates/swc_ecma_minifier/tests/fixture/issues/9460/input.js

```diff
@@ -0,0 +1,3 @@
+const a = 1;
+const { b = "b", c, d } = {};
```

Comment:
Even with `pure_getters`, the computation of the default value for assignment may still have side effects.
For example:

```JavaScript
const { b = console.log("b"), c, d } = {};
```

@Austaras What do you think?

Reply:
`ignore_return_value` will check if the expr is side-effect-free.

Reply:
> `ignore_return_value` will check if the expr is side-effect-free.

It seems that `take_ident_of_pat_if_unused` will return before calling `ignore_return_value`, so that the expression will always be removed if unused.

Reply:
> `ignore_return_value` will check if the expr is side-effect-free.

No. There're multiple places who would call `take_pat_if_unused` like `drop_unused_arrow_params`

Reply:
Maybe just check that assign value is side effect free?

Reply:
> may_have_side_effects

I tried it. But expressions with pure annotation cannot be recognized so that there're still side-effect-free expressions in output code.

**Input code**
```js
const { a = /*#__PURE__*/ sideEffectFree(), b = "b" } = {};
```

**Output code**
```js
const { a = /*#__PURE__*/ sideEffectFree() } = {};
```

Reply:
See the latest commit, expected output is
```js
const { a = sideEffect(), d } = {};
console.log(d);
```
but got
```js
const { a = /*#__PURE__*/ sideEffectFree(sideEffect()), d } = {};
console.log(d);
```
`is_pure_callee` doesn't consume the pure annotation.

Reply:
It's intentional, and if we want to drop it we can invoke `ignore_return_value` from the method.

Reply:
It seems that `ignore_return_value` doesn't consume pure either. Its return expr is same as input expr.